pkg/haproxy/haproxy.go

```diff
@@ -333,48 +370,48 @@ func (h *HAProxyManager) run() {
 }
 
 // Reload rewrites the configuration and sends a signal to HAProxy to initiate the reload
-func (h *HAProxyManager) Reload(ports []uint16) error {
+func (h *HAProxyManager) Reload(podIPs []string, targetPort, servicePort string) error {
 	// compare ports and do nothing if they are the same
-	if reflect.DeepEqual(ports, h.ports) {
+	if targetPort == h.targetPort && servicePort == h.servicePort && reflect.DeepEqual(podIPs, h.podIPs) {
 		return nil
 	}
 
 	// render template
-	b, err := h.render(ports)
+	b, err := h.render(podIPs, targetPort, servicePort)
 	if err != nil {
-		return fmt.Errorf("error rendering configuration. s=%s d=%v p=%v. %v", h.listenAddr, h.serviceAddrs, ports, err)
+		return fmt.Errorf("error rendering configuration. s=%s d=%v p=%v. %v", h.listenAddr, h.podIPs, targetPort, err)
 	}
 
 	// write template
 	if err := h.write(b); err != nil {
-		return fmt.Errorf("error writing configuration. s=%s d=%v p=%v. %v", h.listenAddr, h.serviceAddrs, ports, err)
+		return fmt.Errorf("error writing configuration. s=%s d=%v p=%v. %v", h.listenAddr, h.podIPs, targetPort, err)
 	}
 
 	// reload haproxy
 	if err := h.reload(); err != nil {
 		// if things go wrong, unroll the write
 		h.unroll()
-		return fmt.Errorf("unable to reload haproxy. s=%s d=%v p=%v. %v", h.listenAddr, h.serviceAddrs, ports, err)
+		return fmt.Errorf("unable to reload haproxy. s=%s d=%v p=%v. %v", h.listenAddr, h.podIPs, targetPort, err)
 	}
 
 	h.rendered = b
-	h.ports = ports
+	h.targetPort = targetPort
```

Comment:
should this also set `servicePort` and `podIPs`?  those are used in the admission/equality comparison at the beginning of the function.

Reply:
I believe it should. Done